### ISSUE TITLE
Optimize favicon bug

### ISSUE THREAD

**Issue by uPagge** [squidfunk/mkdocs-material#5098]
### Bug description

The optimization plugin does not work, it crashes with an error at startup:

WARNING  -  Config value 'dev_addr': The use of the IP address '0.0.0.0' suggests a production environment or the use of a proxy to connect to the MkDocs server.
            However, the MkDocs' server is intended for local development purposes only. Please use a third party production-ready server instead.
INFO     -  Building documentation...
INFO     -  Cleaning site directory

  Optimizations:
FileNotFoundError: [Errno 2] No such file or directory: '.cache/plugin/optimize/images/assets/images/favicon.png'

Adding favicon to the assets/images/favicon.png folder does not solve the problem.

Only adding an image to '.cache/plugin/optimize/images/assets/images/favicon.png' Fixes this problem, but it's a clear crutch.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Building an optimized site](https://squidfunk.github.io/mkdocs-material/setup/building-an-optimized-site/?h=optimi#building-an-optimized-site)

### Reproduction

[en.zip](https://github.com/squidfunk/mkdocs-material/files/10831277/en.zip)

### Steps to reproduce

It is sufficient to simply run the assembly

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by uPagge** [comment]
Hmm, it doesn't seem to be about the favicon. It's just that I'm getting an error on any image. I added an image to the article and got

  Optimizations:
FileNotFoundError: [Errno 2] No such file or directory: '.cache/plugin/optimize/images/features/notify/notify-new-mr.png'

**Comment by squidfunk** [comment]
Thanks for reporting. I've downloaded and run your example – I see no errors:

```
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: /Users/squidfunk/Downloads/en/site

  Optimizations:
    *.png 1.9 kB ↓ 0.0 B [0.0%]

INFO     -  Documentation built in 0.38 seconds
```

What's missing from your example is necessary additional information that is all collected by the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) – installed dependencies, runtime info, etc. Thanks for attaching the reproduction, but it is incomplete without this information.

Thus, without the additional info, we're sadly unable to help you. If you provide a complete reproduction, we can look into it again, but until then, we're closing this issue as not reproducible.

**Comment by yrashk** [comment]
👍 I am also having this problem, will try to reproduce

**Comment by yrashk** [comment]
I am not sure about minimal reproduction, but I've traced it down to being triggered during the printing of the gain summary. Disabling it works for now:

```
plugins:
  - optimize:
         print_gain_summary: false
```

**Comment by squidfunk** [comment]
@yrashk that is weird – if you could pack up a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/), I'd be happy to investigate!

**Comment by JuLeeAtPlanet** [comment]
@yrashk 's fix work-around worked for me.

**Comment by squidfunk** [comment]
This should be fixed now, see https://github.com/squidfunk/mkdocs-material/issues/5676#issuecomment-1623426578. It's likely that `pngquant` is not installed on the system.

**Comment by squidfunk** [comment]
Resolved via #5667, see https://github.com/squidfunk/mkdocs-material/issues/5676#issuecomment-1623715504

**Comment by squidfunk** [comment]
Released as part of [9.1.18+insiders-4.37.0](https://github.com/squidfunk/mkdocs-material-insiders/releases/tag/9.1.18-insiders-4.37.0) – I'm sufficiently confident that this should fix all reported problems, i.e., instruct the user to install `pngquant` and `pillow`, if they're not available.
